### PR TITLE
Added pages for office CAT A Mod 2 MES-4439

### PR DESCRIPTION
**Pull request by edwardwilson**
## Description
Added new pages for CAT A Mod2 office page
## Checklist

- [ ] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [ ] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/pages/office/cat-a-mod2/office.cat-a-mod2.page.ts [hunk 1]
```diff
@@ -0,0 +1,607 @@
...
+import { EyesightTestAddComment } from '../../../modules/tests/test-data/common/eyesight-test/eyesight-test.actions';
+import { CommentSource, FaultSummary } from '../../../shared/models/fault-marking.model';
+import { OutcomeBehaviourMapProvider } from '../../../providers/outcome-behaviour-map/outcome-behaviour-map';
+import { behaviourMap } from '../office-behaviour-map.cat-be';
+import { ActivityCodeModel, activityCodeModelList } from '../components/activity-code/activity-code.constants';
+import { CompetencyOutcome } from '../../../shared/models/competency-outcome';
+import { startsWith } from 'lodash';
+import { getRekeyIndicator } from '../../../modules/tests/rekey/rekey.reducer';
+import { isRekey } from '../../../modules/tests/rekey/rekey.selector';
+import { CAT_BE, JOURNAL_PAGE } from '../../page-names.constants';
```
Comment (mgoldson) [inline src/pages/office/cat-a-mod2/office.cat-a-mod2.page.ts]: TODO here?

File: src/pages/office/cat-a-mod2/office.cat-a-mod2.page.ts [hunk 2]
```diff
@@ -0,0 +1,607 @@
...
+      displayDrivingFault$: currentTest$.pipe(
+        select(getTestOutcome),
+        withLatestFrom(currentTest$.pipe(
+          select(getTestData),
+        )),
+        map(([outcome, data]) =>
+          this.outcomeBehaviourProvider.isVisible(
+            outcome,
+            'faultComment',
+            this.faultSummaryProvider.getDrivingFaultsList(data, TestCategory.BE),
```
Comment (mgoldson) [inline src/pages/office/cat-a-mod2/office.cat-a-mod2.page.ts:315]: TODO here?

File: src/pages/office/cat-a-mod2/office.cat-a-mod2.page.ts [hunk 3]
```diff
@@ -0,0 +1,607 @@
...
+      const controlOrObservation = segments[2];
+      this.store$.dispatch(
+        new AddManoeuvreComment(
+          fieldName,
+          CompetencyOutcome.D,
+          controlOrObservation,
+          dangerousFaultComment.comment),
+      );
+
+    } else if (dangerousFaultComment.source === CommentSource.UNCOUPLE_RECOUPLE) {
```
Comment (mgoldson) [inline src/pages/office/cat-a-mod2/office.cat-a-mod2.page.ts:473]: Are we able to remove anything to do with Uncouple/Recouple at this stage?

**Review by mgoldson — COMMENTED** [review]
A few TODO's to add and a question.

File: src/pages/office/cat-a-mod2/office.cat-a-mod2.page.ts (hunk 3 above)
Comment (edwardwilson) [reply src/pages/office/cat-a-mod2/office.cat-a-mod2.page.ts:473]: We try to leave functional work for the later stories and just creat the pages now

File: src/pages/office/cat-a-mod2/office.cat-a-mod2.page.ts (hunk 1 above)
Comment (szabikr) [reply src/pages/office/cat-a-mod2/office.cat-a-mod2.page.ts]: Actually, changing the page constants

File: src/pages/office/cat-a-mod2/__tests__/office.cat-a-mod2.page.spec.ts [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+        ComponentsModule,
+        StoreModule.forRoot({
+          tests: () => ({
+            currentTest: {
+              slotId: '123',
+            },
+            testStatus: {},
+            startedTests: {
+              123: {
+                category: TestCategory.BE,
```
Comment (szabikr) [inline src/pages/office/cat-a-mod2/__tests__/office.cat-a-mod2.page.spec.ts]: TODO needed

File: src/pages/office/cat-a-mod2/components/independent-driving/independent-driving.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+import {
+  OutcomeBehaviourMapProvider,
+  VisibilityType,
+} from '../../../../../providers/outcome-behaviour-map/outcome-behaviour-map';
+
+@Component({
+  selector: 'independent-driving',
+  templateUrl: 'independent-driving.html',
+})
+export class IndependentDrivingComponent implements OnChanges {
```
Comment (szabikr) [inline src/pages/office/cat-a-mod2/components/independent-driving/independent-driving.ts:13]: Why are we having many IndependentDrivingComponents when they are not category specific?

File: src/pages/office/cat-a-mod2/office.cat-a-mod2.page.ts [hunk 6]
```diff
@@ -0,0 +1,608 @@
...
+  QuestionResult,
+} from '@dvsa/mes-test-schema/categories/common';
+import {
+  AddDangerousFaultComment,
+} from '../../../modules/tests/test-data/common/dangerous-faults/dangerous-faults.actions';
+import { AddSeriousFaultComment } from '../../../modules/tests/test-data/common/serious-faults/serious-faults.actions';
+import { AddDrivingFaultComment } from '../../../modules/tests/test-data/common/driving-faults/driving-faults.actions';
+import {
+  AddShowMeTellMeComment,
+} from '../../../modules/tests/test-data/cat-be/vehicle-checks/vehicle-checks.cat-be.action';
```
Comment (szabikr) [inline src/pages/office/cat-a-mod2/office.cat-a-mod2.page.ts:88]: Todo needed

File: src/pages/office/cat-a-mod2/office.cat-a-mod2.page.ts [hunk 7]
```diff
@@ -0,0 +1,608 @@
...
+import { AddManoeuvreComment } from '../../../modules/tests/test-data/common/manoeuvres/manoeuvres.actions';
+import { EyesightTestAddComment } from '../../../modules/tests/test-data/common/eyesight-test/eyesight-test.actions';
+import { CommentSource, FaultSummary } from '../../../shared/models/fault-marking.model';
+import { OutcomeBehaviourMapProvider } from '../../../providers/outcome-behaviour-map/outcome-behaviour-map';
+import { behaviourMap } from '../office-behaviour-map.cat-be';
```
Comment (szabikr) [inline src/pages/office/cat-a-mod2/office.cat-a-mod2.page.ts:94]: Todo needed

**Review by szabikr — APPROVED** [review]
A few Todos are missing

**Comment by mgoldson** [comment]
@edwardwilson Happy to approve once you've addressed the comments from Szabi 😀

File: src/pages/office/cat-a-mod2/components/independent-driving/independent-driving.ts (hunk 5 above)
Comment (edwardwilson) [reply src/pages/office/cat-a-mod2/components/independent-driving/independent-driving.ts:13]: True, they have been copied in the past categories. Should we resolve this in a story later? Doesn't feel right to do that when this is about creating placeholder pages.

**Pull request merged**